### ISSUE TITLE
Formalize input preprocessing

### ISSUE THREAD

**Issue by psyhtest** [mlcommons/training_policies#30]
In our experience, input preprocessing may significantly affect accuracy scores.

For example, when evaluating the [Intel Caffe artifact](https://github.com/ctuning/ck-request-asplos18-caffe-intel) submitted to [ReQuEST@ASPLOS'18](http://cknowledge.org/request-cfp-asplos2018.html), we measured the following Top-1/Top-5 ImageNet [scores](https://github.com/ctuning/ck-request-asplos18-caffe-intel/issues/7#issuecomment-367319271) for [ResNet50](https://arxiv.org/abs/1512.03385) in single-precision floating-point:

Resolution | Top-1 | Top-5
------|--------------|-------------
224 | 0.707186 | 0.898327
256 | 0.729173 | 0.911852
320 | 0.731754 | 0.912832
320' | 0.736136 | 0.914713

This table shows that while ResNet50 [operates](https://github.com/KaimingHe/deep-residual-networks/blob/master/prototxt/ResNet-50-deploy.prototxt) on _224x224_ input images, rescaling to a higher resolution and then cropping to _224x224_ can boost the Top-1 accuracy score by up to 3%! 

Furthermore, the accuracy may significantly depend on the mean values used. In the above table, the row indexed _320_ was obtained with the mean calculated over the _validation_ set, while the row indexed _320'_ with the mean calculated over the _training_ set.

It is clear that this or similar preprocessing techniques can in principle be used to hide significant accuracy degradation from aggressive quantization or pruning techniques. Therefore, their use should be formalized.

Unspecified preprocessing could also be the reason why we [struggled](https://github.com/ARM-software/ComputeLibrary/issues/431) to validate the [claimed MobileNets-v1 accuracy](https://github.com/tensorflow/models/blob/1630da3434974e9ad5a0b6d887ac716a97ce03d3/research/slim/nets/mobilenet_v1.md) against [our own implementations](https://github.com/dividiti/ck-request-asplos18-mobilenets-armcl-opencl) using TensorFlow and the Arm Compute Library, with **the same pretrained weights**.

This is relevant not just for inference but for training too. Collecting data about the costs and gains of preprocessing, especially for techniques like [progressive resizing](http://www.fast.ai/2018/04/30/dawnbench-fastai/), could also help autotune the training process.

**NB:** Originally opened as https://github.com/mlperf/reference/issues/48

**Comment by petermattson** [comment]
Combining this with issue 16 (will close as duplicate), which summarizes as: what is legal, what is timed?

**Comment by bitfort** [comment]
- In general, do what the reference implementation does.
- Modifications are allowed for scaling. There will be a forthcoming list of pre-approved modifications. These modifications can be used any system size, if desired.
- When taking timing measurements, measure time the same way the reference implementation does. If the behavior isn't clear, or is undesirable, file an issue and further clarification will be provided. 
- A cheat-sheet will be provided for each model with the most common cases covered.
- As far as caching and system tricks, you should flush the caches prior to each run (or something similar, such as restarting the computer). If you have a special case, file an issue and additional guidance can be discussed and given.

**Comment by psyhtest** [comment]
> Modifications are allowed for scaling. There will be a forthcoming list of pre-approved modifications.

Hmm, but the accuracy is likely to change (as the data above shows). How about calculating a reference accuracy for each pre-approved modification, and ask submitters to adhere to it?

**Comment by petermattson** [comment]
Further discussion in working group: time only preprocessing that must be done for each run in v0.5.
